ckanext/unaids/logic.py

```diff
@@ -35,3 +36,49 @@ def update_filename_in_resource_url(resource):
         new_url_segments = url_segments[:-1] + [filename]
         resource['url'] = '/'.join(new_url_segments)
     return resource
+
+
+def auto_populate_data_dictionary(context, resource_dict):
+    table_schema_name = resource_dict.get('schema')
+
+    if not table_schema_name:
+        return
```

Comment:
In the PR description, you're saying
> The logic fails silently if you try to auto-populate the data dictionary for a resource that has no associated table schema.

I wouldn't call this behaviour "fail silently" but rather bypassed. Did I miss  something?

This `return None` behaviour here might be confusing for API users. How about using `toolkit.get_or_bust(resource_dict, 'schema')` here and handling running the auto population only for schemed resources in the `after_upload` handler:
```
    def after_upload(self, context, resource_dict, dataset_dict):
       if 'schema' in resource_dict:
            try:
                logic.auto_populate_data_dictionary(context, resource_dict)
            except Exception as e:
```

That way your intent is much more explicit and `auto_populate_data_dictionary` errors more friendly.